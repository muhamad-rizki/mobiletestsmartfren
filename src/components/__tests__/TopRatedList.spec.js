/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import {
  TopRatedList,
} from '../index';

describe('TopRatedList Component', () => {
  it('renders as expected', () => {
    const setTRLoading = jest.fn();
    const wrapper = shallow(
      <TopRatedList
        topRated={{
          results: [],
        }}
        imgUrl="https://image.tmdb.org/t/p/"
        topRatedIndex={0}
        setTRLoading={setTRLoading}
        loadingTopRated={false}
        activeTabIndex={1}
        // eslint-disable-next-line jsx-a11y/tabindex-no-positive
        tabIndex={1}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
