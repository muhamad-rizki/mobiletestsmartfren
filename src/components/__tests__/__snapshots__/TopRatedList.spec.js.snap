// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopRatedList Component renders as expected 1`] = `
<ForwardRef
  ListEmptyComponent={[Function]}
  data={Array []}
  initialNumToRender={5}
  keyExtractor={[Function]}
  maxToRenderPerBatch={5}
  onEndReachedThreshold={0.5}
  pinchGestureEnabled={false}
  renderItem={[Function]}
  scrollEnabled={false}
  windowSize={1}
/>
`;

exports[`TopRatedList Component renders as expected with at least one item 1`] = `
<ForwardRef
  ListEmptyComponent={[Function]}
  data={
    Array [
      Object {
        "adult": false,
        "backdrop_path": "/v4yVTbbl8dE1UP2dWu5CLyaXOku.jpg",
        "genre_ids": Array [
          12,
        ],
        "id": 420817,
        "original_language": "en",
        "original_title": "Aladdin",
        "overview": "A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground 'fight clubs' forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.",
        "popularity": 128.906,
        "poster_path": "/3iYQTLGoy7QnjcUYRJy4YrAgGvp.jpg",
        "release_date": "2019-05-22",
        "title": "Aladdin",
        "video": false,
        "vote_average": 7.2,
        "vote_count": 1950,
      },
    ]
  }
  initialNumToRender={5}
  keyExtractor={[Function]}
  maxToRenderPerBatch={5}
  onEndReachedThreshold={0.5}
  pinchGestureEnabled={false}
  renderItem={[Function]}
  scrollEnabled={false}
  windowSize={1}
/>
`;
